Input processor VerifyName removal and unordered map additions

pshires opened the pull request:
Remove VerifyName calls in favor of unordered map lookups when verifying names between different E+ object types. VerifyName is not needed when verifying within the same E+ object type, as the JSON format takes care of this already. This gives amortized O(1) time to verify a name instead of the original linear O(n) time to verify a name. 
### Work Checklist

Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
- [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
- [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
  - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
  - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
  - NewFeature: This pull request includes code to add a new feature to EnergyPlus
  - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
  - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog
### Review Checklist

This will not be exhaustively relevant to every PR.
- [ ] Code style (parentheses padding, variable names)
- [ ] Functional code review (it has to work!)
- [ ] If defect, results of running current develop vs this branch should exhibit the fix
- [ ] CI status: all green or justified
- [ ] Performance: CI Linux results include performance check -- verify this
- [ ] Unit Test(s)
- C++ checks:
  - [ ] Argument types
  - [ ] If any virtual classes, ensure virtual destructor included, other things
- IDD changes:
  - [ ] Verify naming conventions and styles, memos and notes and defaults
  - [ ] Open windows IDF Editor with modified IDD to check for errors
  - [ ] If transition, add rules to spreadsheet
  - [ ] If transition, add transition source
  - [ ] If transition, update idfs
- [ ] If new idf included, locally check the err file and other outputs
- [ ] Documentation changes in place
- [ ] Changed docs build successfully
- [ ] ExpandObjects changes?
- [ ] If output changes, including tabular output structure, add to output rules file for interfaces

Commit:
VerifyUnique..Name() refactoring

Changed data structure for objects that have to be unique among all
object of that object type. We use hash map instead of array now. In
VerifyUniqueName() routines we don’t do linear search and unnecessary
allocations any more, instead use log(N) find method.

Commit:
Removed unnecessary VerifyName()

In some parts of the source file we had unnecessary VerifyName() calls,
since json() takes care of duplicates we could remove it. There is 1
place in ZoneTempPredictorCorrector.cc where we have to verify for
duplicates among Names and ZoneNames, and another place where we have
to verify for duplicates among SingleCooling and SingleHeatingOrCooling
objects. I kept Verifyname there. Search algorithm might needed to be
re-thought.

Commit:
Removed duplicated FindItemInList()

Commit:
Removed unnecessary VerifyName() call

json() takes care of duplicates in this case

pshires committed:
Remove intra-object verifyName calls, add unordered_set to verify for one particular object

pshires committed:
Merge remote-tracking branch 'origin/verify_unique_names' into input_processor_verify_name_experiment

pshires committed:
Finish editing the B's in the source files, and merge Ksenia's branch with mine

pshires committed:
Done with the C's, Curve objects now check an unordered_map for unique names

pshires committed:
Remove a couple more VerifyName calls

Commit:
Removed unnecessary verifyName() calls

Commit:
VerifyName() removal

Commit:
Removed unnecessary VerifyName() checks

* fix files starting from the bottom of the list of all source files
* stopped on “OutsideEnergySources.cc” (exclusive)

pshires committed:
Add a couple unordered maps for log time Name verification, and a function to do the verification itself. Remove VerifyName calls, mark with unique_string_blank_key comments to be able to quickly find spots to double check empty string names in the future

Commit:
Removed redundant code

* removed unnecessary verifyName() calls
* added functions to check string for emptiness
*removed redundant array allocations

Commit:
Removed redundant code

* verifyName()
*IsNameEmpty() in replace of if-else

pshires committed:
Remove more VerifyNames and add more unordered maps, clear_state functions, and relocate a couple static function variables to their respective namespace level

pshires committed:
Remove even more VerifyNames and add more unordered maps and relocate a more static function variables to their respective namespace levels :)

pshires committed:
Call FluidCooler::clear_state in the EnergyPlusFixture

Commit:
Continued verifyName cleanUp

pshires committed:
Remove VerifyName calls

pshires committed:
add a unique_string to replace later with a function to check for blank names

pshires committed:
cleaup. mostly indention cleanup caused by Clion inserting spaces for tabs

pshires committed:
indention cleanup

Commit:
Last VerifyName() fixes

pshires committed:
attempt vim full indent of the file for cleanup

Commit:
Fixed Indenting

Commit:
Fixed indenting

pshires committed:
Revert "attempt vim full indent of the file for cleanup"

This reverts commit 071a36fbfa138a0656c91054c61a61b50a661dcd.

Every line was considered changed which doesn't go over well when viewing changes in a PR on Github

pshires committed:
Merge remote-tracking branch 'origin/verify_unique_names' into verify_unique_names

Commit:
Fixed Indenting

Commit:
Revert "Fixed Indenting"

This reverts commit 8589e2cdbf4ddbe3819895f925c4e504e894f989.

Commit:
Fixed indenting

pshires committed:
cleanup indentions

pshires committed:
Merge remote-tracking branch 'origin/verify_unique_names' into verify_unique_names

Commit:
Removed TODOs

Commit:
fixed indenting

Commit:
removed todo

pshires committed:
Merge remote-tracking branch 'origin/input_processor_verify_name_experiment' into verify_unique_names

# Conflicts:
#	src/EnergyPlus/ZoneTempPredictorCorrector.cc

pshires committed:
Merge remote-tracking branch 'origin/verify_unique_names' into verify_unique_names

# Conflicts:
#	src/EnergyPlus/ZoneTempPredictorCorrector.cc

pshires committed:
Remove TODO (forgot to pull that change before merging)

pshires committed:
Replace the unique string with IsNameEmpty calls( through letter C ), to keep the same error functionality of the original input processor

mbadams5 commented on src/EnergyPlus/AirflowNetworkBalanceManager.cc:
```diff
@@ -2435,36 +2357,28 @@ namespace AirflowNetworkBalanceManager {
-						if ( IsBlank ) Alphas( 5 ) = "xxxxx";
+					InputProcessor::VerifyUniqueInterObjectName( UniqueAirflowNetworkSurfaceName, Alphas( 1 ), CurrentModuleObject, ErrorsFound );
+					if ( ErrorsFound ) {
+						if (IsBlank) Alphas(5) = "xxxxx";
```
This should be taken care of within `VerifyUniqueInterObjectName`.

mbadams5 commented on src/EnergyPlus/AirflowNetworkBalanceManager.cc:
```diff
@@ -686,14 +688,7 @@ namespace AirflowNetworkBalanceManager {
-					if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-				}
-
+				InputProcessor::IsNameEmpty( Alphas( 1 ), CurrentModuleObject + " Name", "xxxxx", ErrorsFound );
```
Why pass in `"xxxxx"`? It is just to set the name to something different so we can just do it within `IsNameEmpty()`. Also, why create a temporary string to pass into the function `CurrentModuleObject + " Name"`, we can just always append "Name" in the error message in the function and only pass in `CurrentModuleObject`.

mbadams5 commented on src/EnergyPlus/AirflowNetworkBalanceManager.cc:
```diff
@@ -2435,36 +2357,28 @@ namespace AirflowNetworkBalanceManager {
-					if ( IsNotOK ) {
-						ErrorsFound = true;
-						if ( IsBlank ) Alphas( 5 ) = "xxxxx";
+					InputProcessor::VerifyUniqueInterObjectName( UniqueAirflowNetworkSurfaceName, Alphas( 1 ), CurrentModuleObject, ErrorsFound );
```
You are checking the wrong thing here. It should be `Alphas( 5 )`, not `Alphas( 1 )`.

mbadams5 commented on src/EnergyPlus/BoilerSteam.cc:
```diff
@@ -328,14 +326,7 @@ namespace BoilerSteam {
-				ErrorsFound = true;
-				if ( IsBlank ) cAlphaArgs( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty( cAlphaArgs( 1 ), cCurrentModuleObject + " Name", "xxxxx", ErrorsFound );
```
This needs to be `VerifyUniqueInterObjectName` since it is checking names against `Boiler` which the `Boilers.cc` file does as well.

mbadams5 commented on src/EnergyPlus/Boilers.cc:
```diff
@@ -344,14 +342,7 @@ namespace Boilers {
-				ErrorsFound = true;
-				if ( IsBlank ) cAlphaArgs( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty( cAlphaArgs( 1 ), cCurrentModuleObject + " Name", "xxxxx", ErrorsFound );
```
Needs to be `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/BranchInputManager.cc:
```diff
@@ -1447,23 +1445,12 @@ namespace BranchInputManager {
-						}
-					}
+					bool is_blank = false;
+					InputProcessor::IsNameEmpty( Alphas( 1 ), CurrentModuleObject + " Name", is_blank, ErrFound );
```
Maybe it would be better to return a bool instead of overloading and passing in by reference.

mbadams5 commented on src/EnergyPlus/CondenserLoopTowers.cc:
```diff
@@ -214,6 +214,7 @@ namespace CondenserLoopTowers {
 	Array1D< TowerInletConds > SimpleTowerInlet; // inlet conditions
 	Array1D< ReportVars > SimpleTowerReport; // report variables
 	Array1D< VSTowerData > VSTower; // model coefficients and specific variables for VS tower
+	std::unordered_map< std::string, std::string > SimpleTower_map;
```
This should have a better name and not mix casing.

mbadams5 commented on src/EnergyPlus/CondenserLoopTowers.cc:
```diff
@@ -531,13 +533,8 @@ namespace CondenserLoopTowers {
-				if ( IsBlank ) AlphArray( 1 ) = "xxxxx";
-			}
+			InputProcessor::VerifyUniqueInterObjectName( SimpleTower_map, AlphArray( 1 ), cCurrentModuleObject, ErrorsFound );
+			InputProcessor::IsNameEmpty( AlphArray( 1 ), cCurrentModuleObject + " Name", "xxxxx", ErrorsFound );
```
Empty name check can be done in `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/CurveManager.cc:
```diff
@@ -392,6 +395,22 @@ namespace CurveManager {
 	}
 
 	void
+	VerifyUniqueCurveName(
```
Should probably convert to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/DXCoils.cc:
```diff
@@ -1113,13 +1111,7 @@ namespace DXCoils {
-				ErrorsFound = true;
-				if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-			}
+			// unique_string_blank_key
```
Need to convert all of these to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/DataSurfaceLists.cc:
```diff
@@ -220,14 +218,7 @@ namespace DataSurfaceLists {
-					ErrorsFound = true;
-					if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-				}
+				// unique_string_blank_key
```
Need to convert these to `IsNameEmpty`

mbadams5 commented on src/EnergyPlus/DataZoneEquipment.cc:
```diff
@@ -517,14 +517,7 @@ namespace DataZoneEquipment {
-					if ( IsBlank ) AlphArray( 1 ) = "xxxxx";
-				}
-
+				// unique_string_blank_key_check
```
Need to convert with existing error message.

mbadams5 commented on src/EnergyPlus/DemandManager.cc:
```diff
@@ -148,6 +148,7 @@ namespace DemandManager {
 	// Object Data
 	Array1D< DemandManagerListData > DemandManagerList;
 	Array1D< DemandManagerData > DemandMgr;
+	std::unordered_map< std::string, std::string > DemandMgr_map;
```
Need a better name

mbadams5 commented on src/EnergyPlus/DesiccantDehumidifiers.cc:
```diff
@@ -204,6 +204,7 @@ namespace DesiccantDehumidifiers {
 
 	// Object Data
 	Array1D< DesiccantDehumidifierData > DesicDehum;
+	std::unordered_map< std::string, std::string > DesicDehum_map;
```
Need better name

mbadams5 commented on src/EnergyPlus/DualDuct.cc:
```diff
@@ -154,6 +154,7 @@ namespace DualDuct {
 
 	// Object Data
 	Array1D< DamperDesignParams > Damper;
+	std::unordered_map< std::string, std::string > Damper_map;
```
Need better name

mbadams5 commented on src/EnergyPlus/EvaporativeCoolers.cc:
```diff
@@ -182,6 +182,7 @@ namespace EvaporativeCoolers {
 
 	// Object Data
 	Array1D< EvapConditions > EvapCond;
+	std::unordered_map< std::string, std::string > EvapCond_map;
```
Need better name

mbadams5 commented on src/EnergyPlus/EvaporativeCoolers.cc:
```diff
@@ -3482,13 +3461,8 @@ namespace EvaporativeCoolers {
-					ErrorsFound = true;
-					if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-				}
+				InputProcessor::VerifyUniqueInterObjectName( EvapCond_map, Alphas( 1 ), CurrentModuleObject, ErrorsFound );
```
Should this be here? I don't think so.

mbadams5 commented on src/EnergyPlus/EvaporativeFluidCoolers.cc:
```diff
@@ -155,6 +155,7 @@ namespace EvaporativeFluidCoolers {
 	int const EvapFluidCooler_TwoSpeed( 2 );
 
 	static std::string const BlankString;
+	bool GetInput( true );
```
Make this a more unique name to prevent future name collisions

mbadams5 commented on src/EnergyPlus/ExteriorEnergyUse.cc:
```diff
@@ -118,6 +118,7 @@ namespace ExteriorEnergyUse {
 	int const DistrictHeatUse( 12 ); // Purchased Heating
 	int const OtherFuel1Use( 13 ); // OtherFuel1
 	int const OtherFuel2Use( 14 ); // OtherFuel2
+	bool GetInputFlag( true ); // First time, input is "gotten"
```
Make this a more unique name to prevent future name collisions

mbadams5 commented on src/EnergyPlus/FluidCoolers.cc:
```diff
@@ -140,6 +140,7 @@ namespace FluidCoolers {
 	std::string const cFluidCooler_SingleSpeed( "FluidCooler:SingleSpeed" );
 	std::string const cFluidCooler_TwoSpeed( "FluidCooler:TwoSpeed" );
 
+	bool GetInput( true );
```
Make this a more unique name to prevent future name collisions

mbadams5 commented on src/EnergyPlus/FluidCoolers.cc:
```diff
@@ -182,6 +183,7 @@ namespace FluidCoolers {
 
 	// Object Data
 	Array1D< FluidCoolerspecs > SimpleFluidCooler; // dimension to number of machines
+	std::unordered_map< std::string, std::string > SimpleFluidCooler_map;
```
Change name

mbadams5 commented on src/EnergyPlus/ExteriorEnergyUse.cc:
```diff
@@ -137,6 +138,7 @@ namespace ExteriorEnergyUse {
 	// Object Data
 	Array1D< ExteriorLightUsage > ExteriorLights; // Structure for Exterior Light reporting
 	Array1D< ExteriorEquipmentUsage > ExteriorEquipment; // Structure for Exterior Equipment Reporting
+	std::unordered_map< std::string, std::string > ExteriorEquipment_map;
```
change name

mbadams5 commented on src/EnergyPlus/EvaporativeFluidCoolers.cc:
```diff
@@ -193,6 +194,7 @@ namespace EvaporativeFluidCoolers {
 
 	// Object Data
 	Array1D< EvapFluidCoolerspecs > SimpleEvapFluidCooler; // dimension to number of machines
+	std::unordered_map< std::string, std::string > SimpleEvapFluidCooler_map;
```
better name

mbadams5 commented on src/EnergyPlus/Fans.cc:
```diff
@@ -196,6 +196,7 @@ namespace Fans {
 
 	// Object Data
 	Array1D< FanEquipConditions > Fan;
+	std::unordered_map< std::string, std::string > Fan_map;
```
better name

mbadams5 commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -266,6 +267,7 @@ namespace Furnaces {
 
 	// Object Data
 	Array1D< FurnaceEquipConditions > Furnace;
+	std::unordered_map< std::string, std::string > Furnace_map;
```
better name

mbadams5 commented on src/EnergyPlus/Furnaces.cc:
```diff
@@ -221,6 +221,7 @@ namespace Furnaces {
 	int const DehumidControl_CoolReheat( 2 );
 
 	static std::string const fluidNameSteam( "STEAM" );
+	bool GetInputFlag( true ); // Logical to allow "GetInput" only once per simulation
```
Make this a more unique name to prevent future name collisions

mbadams5 commented on src/EnergyPlus/HVACHXAssistedCoolingCoil.cc:
```diff
@@ -512,13 +506,8 @@ namespace HVACHXAssistedCoolingCoil {
-				ErrorsFound = true;
-				if ( IsBlank ) AlphArray( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty(AlphArray( 1 ), CurrentModuleObject + " Name", "xxxxx", ErrorsFound);
```
This should use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/HVACHXAssistedCoolingCoil.cc:
```diff
@@ -391,15 +391,9 @@ namespace HVACHXAssistedCoolingCoil {
-				if ( IsBlank ) AlphArray( 1 ) = "xxxxx";
-			}
-			HXAssistedCoil( HXAssistedCoilNum ).Name = AlphArray( 1 );
+			InputProcessor::IsNameEmpty(AlphArray( 1 ), CurrentModuleObject + " Name", "xxxxx", ErrorsFound);
```
This should use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/HeatingCoils.cc:
```diff
@@ -460,13 +460,7 @@ namespace HeatingCoils {
-				InputErrorsFound = true;
-				if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty(Alphas( 1 ), CurrentModuleObject + " Name", "xxxxx", InputErrorsFound);
```
This needs to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/HeatingCoils.cc:
```diff
@@ -525,13 +519,7 @@ namespace HeatingCoils {
-				InputErrorsFound = true;
-				if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty(Alphas( 1 ), CurrentModuleObject + " Name", "xxxxx", InputErrorsFound);
```
This needs to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/HeatingCoils.cc:
```diff
@@ -600,13 +588,7 @@ namespace HeatingCoils {
-				InputErrorsFound = true;
-				if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty(Alphas( 1 ), CurrentModuleObject + " Name", "xxxxx", InputErrorsFound);
```
This needs to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/HeatingCoils.cc:
```diff
@@ -686,13 +668,7 @@ namespace HeatingCoils {
-				InputErrorsFound = true;
-				if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty(Alphas( 1 ), CurrentModuleObject + " Name", "xxxxx", InputErrorsFound);
```
This needs to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/HeatingCoils.cc:
```diff
@@ -777,13 +753,7 @@ namespace HeatingCoils {
-				InputErrorsFound = true;
-				if ( IsBlank ) Alphas( 1 ) = "xxxxx";
-			}
+			InputProcessor::IsNameEmpty(Alphas( 1 ), CurrentModuleObject + " Name", "xxxxx", InputErrorsFound);
```
This needs to use `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/InputProcessor.cc:
```diff
@@ -1649,6 +1653,22 @@ namespace EnergyPlus {
 	}
 
+    void
+    InputProcessor::VerifyUniqueInterObjectName(
```
I think this should be in `GlobalNames.hh`

mbadams5 commented on src/EnergyPlus/InputProcessor.cc:
```diff
@@ -1649,6 +1653,22 @@ namespace EnergyPlus {
+			bool & ErrorsFound
+    ) {
+		auto const & names_iter = names.find( object_name );
+        if ( names_iter == names.end() ) {
```
Need to fix indention

mbadams5 commented on src/EnergyPlus/InputProcessor.cc:
```diff
@@ -1649,6 +1653,22 @@ namespace EnergyPlus {
+			names.emplace( object_name, object_type );
+		} else {
+			ErrorsFound = true;
+			ShowSevereError( object_name + " with object type " + object_type + " duplicates a name in object type " + names_iter->second );
```
Might need more error messaging for specific objects.

mbadams5 commented on src/EnergyPlus/InputProcessor.hh:
```diff
@@ -802,17 +814,40 @@ namespace EnergyPlus {
-			Optional_bool_const UpperBoundCondition = _, // Condition for error condition, if applicable
-			Optional_string_const ValueString = _, // Value with digits if to be displayed with error
-			Optional_string_const WhatObjectName = _ // ObjectName -- used for error messages
+				bool & ErrorsFound, // Set to true if error detected
```
Why the indention change?

kburova replied:
This is some kind of bug. There is nothing wrong with indention. I checked everything around those lines.

mbadams5 commented on src/EnergyPlus/InputProcessor.hh:
```diff
@@ -23,6 +24,7 @@
 // EnergyPlus Headers
 #include <EnergyPlus.hh>
 #include <DataGlobals.hh>
+#include <GlobalNames.hh>
```
Why include?

kburova replied:
Project doesn't build without it. There are files where we use InputProcessor:: we also use data structures form GlobalNames. There are more then one file. So if we want to remove this library from InputProcessor we will have to go through all those files that use Data Structs from GlobalNames and add that library.

mbadams5 commented on src/EnergyPlus/InputProcessor.hh:
```diff
@@ -10,6 +10,7 @@
 #include <unordered_map>
 #include <unordered_set>
 #include <fstream>
+#include <set>
```
Why include?

mbadams5 commented on src/EnergyPlus/InputProcessor.cc:
```diff
@@ -2,13 +2,15 @@
 #include <algorithm>
 #include <istream>
 #include <iostream>
+#include <set>
```
Why include?

mbadams5 commented on src/EnergyPlus/InputProcessor.cc:
```diff
@@ -2,13 +2,15 @@
 #include <ObjexxFCL/Optional.hh>
 
 // EnergyPlus Headers
+#include <GlobalNames.hh>
```
Why include?

mbadams5 commented on src/EnergyPlus/InputProcessor.cc:
```diff
@@ -20,6 +22,7 @@
 #include <SortAndStringUtilities.hh>
 #include <milo/dtoa.hpp>
 #include <milo/itoa.hpp>
+#include <mach/mach_types.h>
```
Why include?

mbadams5 commented on src/EnergyPlus/Photovoltaics.cc:
```diff
@@ -401,12 +401,10 @@ namespace Photovoltaics {
-			IsBlank = false;
-			InputProcessor::VerifyName( cAlphaArgs( 1 ), PVarray, PVnum - 1, IsNotOK, IsBlank, cCurrentModuleObject + " Name" );
-			if ( IsNotOK ) {
+			if ( cAlphaArgs( 1 ).empty() ) {
```
Convert to use `IsNameEmpty`

mbadams5 commented on src/EnergyPlus/Photovoltaics.cc:
```diff
@@ -531,10 +529,8 @@ namespace Photovoltaics {
-				IsBlank = false;
-				InputProcessor::VerifyName( cAlphaArgs( 1 ), tmpSimpleModuleParams, ModNum - 1, IsNotOK, IsBlank, cCurrentModuleObject + " Name" );
-				if ( IsNotOK ) { //repeat or blank name so don't add
+				if ( cAlphaArgs( 1 ).empty() ) {
```
Convert to use `IsNameEmpty`

mbadams5 commented on src/EnergyPlus/Photovoltaics.cc:
```diff
@@ -575,11 +571,8 @@ namespace Photovoltaics {
-				IsBlank = false;
-				InputProcessor::VerifyName( cAlphaArgs( 1 ), tmpTNRSYSModuleParams, ModNum - 1, IsNotOK, IsBlank, cCurrentModuleObject + " Name" );
-				if ( IsNotOK ) { //repeat or blank name so don't add
+				if ( cAlphaArgs( 1 ).empty() ) {
```
Convert to use `IsNameEmpty`

mbadams5 commented on src/EnergyPlus/Photovoltaics.cc:
```diff
@@ -630,15 +623,11 @@ namespace Photovoltaics {
-				IsBlank = false;
-				InputProcessor::VerifyName( cAlphaArgs( 1 ), tmpSNLModuleParams, &SNLModuleParamsStuct::name, ModNum - 1, IsNotOK, IsBlank, cCurrentModuleObject + " Name" );
-				if ( IsNotOK ) { //repeat or blank name so don't add
+				if ( cAlphaArgs( 1 ).empty() ) {
```
Convert to use `IsNameEmpty`

mbadams5 commented on src/EnergyPlus/PlantCondLoopOperation.cc:
```diff
@@ -443,51 +443,29 @@ namespace PlantCondLoopOperation {
-			Num = 0;
+
+
 			for ( OpNum = 1; OpNum <= NumPlantOpSchemes; ++OpNum ) {
```
Should change indention since you removed outer scope

mbadams5 commented on src/EnergyPlus/SimAirServingZones.cc:
```diff
@@ -197,6 +197,7 @@ namespace SimAirServingZones {
 	int const Fan_ComponentModel( 25 ); // cpw22Aug2010 (new)
 	int const DXHeatPumpSystem( 26 );
 	int const CoilUserDefined( 27 );
+	bool IsNotOK;
```
Why move this to namespace scope?

mbadams5 commented on src/EnergyPlus/SizingManager.cc:
```diff
@@ -1432,13 +1417,8 @@ namespace SizingManager {
-						if ( IsBlank ) cAlphaArgs( 1 ) = "xxxxx";
-					}
+
+					InputProcessor::IsNameEmpty( cAlphaArgs(1), cCurrentModuleObject + " Name", "xxxxx", ErrorsFound);
```
This one looks odd. Just double check that it is only checking within the same object and we can remove the VerifyName call and don't need `VerifyUniqueInterObjectName`

mbadams5 commented on src/EnergyPlus/SteamBaseboardRadiator.cc:
```diff
@@ -391,13 +391,6 @@ namespace SteamBaseboardRadiator {
-
-			if ( IsNotOK ) {
-				ErrorsFound = true;
-			}
```
Empty name check?

kburova replied:
done

mbadams5 commented on src/EnergyPlus/ThermalChimney.cc:
```diff
@@ -270,17 +270,6 @@ namespace ThermalChimney {
-				} else {
-					cAlphaArgs( 1 ) = cAlphaArgs( 1 ) + "--dup";
-				}
-			}
```
Need IsNameEmpty check

kburova replied:
done

mbadams5 commented on src/EnergyPlus/WindTurbine.cc:
```diff
@@ -352,11 +352,6 @@ namespace WindTurbine {
-
-			if ( IsNotOK ) {
-				ErrorsFound = true;
-			}
```
Need IsNameEmpty check

kburova replied:
done

mbadams5 commented on src/EnergyPlus/ZonePlenum.cc:
```diff
@@ -544,15 +531,6 @@ namespace ZonePlenum {
 				ErrorsFound = true;
 			}
-			if ( NumZoneReturnPlenums > 0 ) { // Check if this zone is also used in another plenum
-				IOStat = InputProcessor::FindItemInList( AlphArray( 2 ), ZoneRetPlenCond, &ZoneReturnPlenumConditions::ZoneName );
```
Can we safely remove all of this? It is obviously checking something?

mbadams5 reviewed (commented):
@pshires @kburova 

I don't know what happened to `SurfaceGeometry.cc` but it says thousands of lines were added and deleted. I wonder if it is a line ending issue or indention issue.

Also you need to fix all the remaining `VerifyName`s that need to use `VerifyUniqueInterObjectName`.

Also there are conflicts that need to be resolved.

Commit:
Changed VerifyName() signature, removed old ones

* fixed all usages in all source files

pshires committed:
Add an intra-object string uniqueness check function, add a unit test to help determine correct usage for this case. And make use of that function

Commit:
Removed redundant declarations

Commit:
Added IsNameEmpty() where VerifyName was removed

pshires committed:
Refactor some existing functions and remove some verifyName calls

pshires committed:
Add overloaded VerifyUniqueName function for cases where a field name isn't passed to GetObjectItem. Remove a few VerifyNames in place of verifyUniqueName

Commit:
Deleted unnecessary declarations.

pshires committed:
Remove some more VerifyName calls in place of VerifyUniqueName

Commit:
Removed redundunt declarations

Commit:
Code clean-up

removed redundant declarations
added missing IsNameEmpty() check

Commit:
Reverted some committed earlier changes

Commit:
Removed a few more redundant declarations

pshires committed:
Continue verifyName removal through the O files

pshires committed:
Merge pull request #3 from ORNL-BTRIC/verify_name_exp_cleanup

Verify name exp cleanup

pshires committed:
Remove VerifyNames through R

Commit:
Removed a few VerifyNames() calls

* added VerifyUnique ones instead

pshires committed:
Keep on with the continuation of the removals of verifyName calls

Commit:
Merge branch 'input_processor_verify_name_experiment' into verify_name_exp_cleanup

Commit:
Removed redundant code

*VerifyNames
*unnecessary declarations

pshires committed:
VerifyName removals

Commit:
Merge input_processor_verify_name_experiement

# Conflicts:
#	src/EnergyPlus/WaterThermalTanks.cc
#	src/EnergyPlus/WaterToAirHeatPump.cc
#	src/EnergyPlus/WaterToAirHeatPumpSimple.cc

Commit:
Fix merge conflicts

# Conflicts:
#	src/EnergyPlus/WaterThermalTanks.cc
#	src/EnergyPlus/WaterToAirHeatPump.cc
#	src/EnergyPlus/WaterToAirHeatPumpSimple.cc

pshires committed:
Merge pull request #4 from ORNL-BTRIC/verify_name_exp_cleanup

Removed a few VerifyNames() calls

pshires committed:
Finish removal of VerifyName calls

pshires committed:
Fix naming issues and potential namespace conflicts

Commit:
Merge branch 'input_processor_verify_name_experiment' into verify_name_exp_cleanup

Commit:
Removed redundancies

kburova committed:
Merge pull request #5 from ORNL-BTRIC/verify_name_exp_cleanup

Verify name exp cleanup

pshires committed:
Convert VerifyUniqueCurve function to VerifyUniqueInterObjectName, move VerifyUniqueInterObjectName to be in GlobalNames

pshires committed:
Blow out SurfaceGeometry.cc in favor of the SurfaceGeometry.cc from input_processor_refactor_verify_name

pshires committed:
fix indention issues

pshires committed:
resolve merge conflicts and update SurfaceGeometry.cc

pshires committed:
clean up variable names

mbadams5 committed:
Final updates
